### PR TITLE
Use k8s name instead of load balancer for dhfind

### PR DESCRIPTION
**Pull request by ischasny**
Use k8s name instead of load balancer for dhfind to see whether that would reduce a number of find timeouts in dev

### CODE REVIEW

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/indexstar/deployment.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ spec:
             - '--backends=http://cali-indexer:3000/'
             - '--backends=http://ago-indexer:3000/'
             - '--backends=http://dhstore.internal.dev.cid.contact/'
-            - '--backends=http://dhfind.internal.dev.cid.contact/'
+            - '--backends=http://dhfind:40080/'
```
Comment (masih) [inline deploy/manifests/dev/us-east-2/tenant/storetheindex/indexstar/deployment.yaml:23]: `dhfind` service is [patched](https://github.com/ipni/storetheindex/blob/main/deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service.yaml#L15) in dev overlay to be an AWS NLB.

If you'd like to experiment with this, I recommend creating a new `Service` object in dev overlay like [this](https://github.com/ipni/storetheindex/blob/main/deploy/manifests/base/storetheindex-single/service.yaml#L15) with type `ClusterIP` and use that.

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service-cluster-ip.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: dhfind
```
Comment (masih) [inline deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service-cluster-ip.yaml:4]: This name is already taken. service names must be unique in a namespace.

**Review by masih — CHANGES_REQUESTED** [review]

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service-cluster-ip.yaml (hunk 2 above)
Comment (ischasny) [reply deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service-cluster-ip.yaml]: Fixed

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service-cluster-ip.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: dhfind-finder
```
Comment (masih) [inline deploy/manifests/dev/us-east-2/tenant/storetheindex/dhfind/service-cluster-ip.yaml:4]:
```suggestion
  name: dhfind-cluster-ip
```

File: deploy/manifests/dev/us-east-2/tenant/storetheindex/indexstar/deployment.yaml [hunk 4]
```diff
@@ -20,7 +20,7 @@ spec:
             - '--backends=http://cali-indexer:3000/'
             - '--backends=http://ago-indexer:3000/'
             - '--backends=http://dhstore.internal.dev.cid.contact/'
-            - '--backends=http://dhfind.internal.dev.cid.contact/'
+            - '--backends=http://dhfind-finder:40080/'
```
Comment (masih) [inline deploy/manifests/dev/us-east-2/tenant/storetheindex/indexstar/deployment.yaml:23]:
```suggestion
            - '--backends=http://dhfind-cluster-ip:40080/'
```

**Review by masih — APPROVED** [review]

**Pull request merged**